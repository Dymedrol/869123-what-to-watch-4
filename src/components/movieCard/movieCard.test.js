import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movieCard.jsx";

const card = {
  previewImage: `img/macbeth.jpg`,
  name: `Movie title`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

it(`Проверка снепшота компонента MovieCard`, () => {
  const tree = renderer
    .create(<MovieCard
      cardData={card}
      onCardHoverHandler={() => {}}
      onMouseEnter={() => {}}
      onMouseLeave={() => {}}
      renderPlayer={() => {}}
      setActiveItem={() => {}}
      removeActiveItem={() => {}}
    />, {
      createNodeMock: () => {
        return {};
      }})
    .toJSON();

  expect(tree).toMatchSnapshot();
});
