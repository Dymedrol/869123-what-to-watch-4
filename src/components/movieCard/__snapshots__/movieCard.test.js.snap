// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка снепшота компонента MovieCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      className="player__video"
      muted={true}
      poster="img/macbeth.jpg"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Movie title
    </a>
  </h3>
</article>
`;
